src/System.Private.CoreLib/shared/System/Number.Formatting.cs

```diff
@@ -250,6 +250,15 @@ internal static partial class Number
         private const int SinglePrecision = 9;
         private const int DoublePrecision = 17;
 
+        // SinglePrecisionCustomFormat and DoublePrecisionCustomFormat are used to ensure that
+        // custom format strings return the same string as in previous releases when the format
+        // would return x digits or less (where x is the value of the corresponding constant).
```

Comment:
This part in particular (returning the same string as before when less than x digits are requested) is also true for standard-numeric format strings. For the standard-formats, we only differ when more than 15 digits are requested and for the default case on `G` and `R`.